hr_timesheet_hours_billed/models/account_analytic_line.py

```diff
@@ -29,3 +29,11 @@ def create(self, vals_list):
             if not vals.get("unit_amount_billed"):
                 vals["unit_amount_billed"] = vals["unit_amount"]
         return super(AccountAnalyticLine, self).create(vals_list)
+
+    def action_approve_records(self):
+        for record in self:
```

Comment:
Why do we need an iteration here? There will be multiple writes.
Using  `self.write({"approved": True})` will be only one operation instead.
Also would be reasonable to update only records that are not approved already. Use "filtered" for that.